lib/index.js

```diff
@@ -245,9 +245,11 @@ function syncObject(toObj, fromObj, options = {}) {
 				typeof toVal !== 'object' ||
 				!toVal ||
 				Array.isArray(toVal) ||
+				toVal instanceof Date ||
```

Comment:
This fix will work, but there's a better way to do this.  This whole conditional can be condensed to `if (isScalar(toVal) || isScalar(fromVal))`.  The `isScalar()` function is specifically designed to handle things like dates that are treated as scalars but show up as objects.  I'm not sure why the original author didn't use it.